http_server.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+import socket
+import email.utils
+import mimetypes
+
+
+def http_server():
+    server_socket = socket.socket(
+        socket.AF_INET,
+        socket.SOCK_STREAM,
+        socket.IPPROTO_IP)
+    server_socket.bind(('127.0.0.1', 50000))
+    server_socket.listen(10)
+    try:
+        while True:
+            conn, addr = server_socket.accept()
+            buffsize = 16
+            msg = ""
+            done = False
+            while not done:
+                part = conn.recv(buffsize)
+                if len(part) < buffsize:
+                    done = True
+                msg = b"{}{}".format(msg, part)
+            out = parse_request(msg)
+            conn.sendall(out)
+            conn.close()
+    except KeyboardInterrupt:
+        server_socket.close()
+
+
+def response_ok(uri):
+    date = email.utils.formatdate(usegmt=True)
+    code = "HTTP/1.1 200 OK"
+    guess = mimetypes.guess_type(uri)[0]
+    content_type = "Content-Type: {}; charset=UTF-8".format(guess)
+    headers = "{}\r\n{}\r\n{}\r\n\r\n".format(code, date, content_type)
```

Comment:
Can you think of a more efficient, more flexible way to accomplish what you are doing here?  What if I want to add a new header?  What if I want to add 5 new headers?